rust/numaflow-core/src/shared/server_info.rs

```diff
@@ -265,7 +265,7 @@ fn trim_after_dash(input: &str) -> &str {
 
 /// Extracts the container type from the server info file.
 /// The file name is in the format of <container_type>-server-info.
-fn get_container_type(server_info_file: &PathBuf) -> Option<&str> {
```

Comment:
just curious, how does this type conversion work and why do we need to make this line of change?

Reply:
The conversion works because of Deref coersion. `PathBuf` implements `Deref` to `Path` https://doc.rust-lang.org/std/path/struct.PathBuf.html#impl-Deref-for-PathBuf (similar to `String` and `str`).
The reason for doing it was that it was showing in `cargo clippy -p numaflow-core`